### ISSUE TITLE
Create central entry point for build documentation

### ISSUE THREAD

**Issue by BenBE** [audacity/audacity#872]
given that building audacity is non-trivial<sup>[[citation needed](https://xkcd.com/285/)]</sup> there should be a central document covering the basics of building the software, required dependencies like build tools and the general overview of things that need to be done, what settings there are available and so on.

Thus it would be great to have a central `BUILDING.md` or similar in the root directory of the source tree that covers the basics of building and referencing the other, platform-specific documentation in `<platform>/build.txt` as appropriate. This central document also should provide references to other important sources of information and where further help can be obtained. The purpose of this central document is to provide all necessary details to building the software so they are available offline given only the source tarball.

the build information also could provide references to how code contributions (`CONTRIBUTIONS.md` and `STYLEGUIDE.md`) should be handled.

And while a Wiki to collect these information may be useful for quick contributions of tips&tricks for the build process it is largely unsatisfactory for someone who has to rely on the bare minimum of external sources.

This issue came about based on missing satisfaction with the overall build and its documentation as discussed in several other issues, like #625, #835, #853.

**Comment by SteveDaulton** [comment]
> Given that building Audacity is non-trivial[[citation needed](https://xkcd.com/285/)] there should be a central document covering the basics of building the software, required dependencies like build tools and the general overview of things that need to be done, what settings there are available and so on.

That would be here: https://wiki.audacityteam.org/wiki/For_Developers
Though it is not always as up to date as we would like.

**Comment by JamesCrook** [comment]
@SteveDaulton - An incorrect explanation for closing.

That link is not in the source tree, which BenBE is asking for in the text, though not spelling out in the issue title.  The correct link for justifying closing is this one, which IS in the source tree:

https://github.com/audacity/audacity/blob/master/INSTALL 

It does reference the other platform-specific docs, and there is enough difference between platforms that it is right that INSTALL delegates everything to those docs.

Our policy is to (try to) provide one way of building, written up in the source tree, and then offer variations (such as WSL, signing on Windows, fetching the manual) in the wiki.  Our how to build documentation needs to be improved.  One approach is to treat the CI builds as the gold standard, since they are always working, and then document and date variations, with a caveat emptor warning, since build instructions, such as our long obsolete instructions for MinGW rot if there is no one actively using that build.

I would be in favour of a new discussion topic on improving build instructions.  'Issues' by their nature need to get closed, whereas discussion topics can be ongoing.  I do think keeping build instructions in a good state is an ongoing discussion, is important, and so many different people have different needs from them.

**Comment by SteveDaulton** [comment]
> @SteveDaulton - An incorrect explanation for closing.
> 
> That link is not in the source tree, which BenBE is asking for in the text, though not spelling out in the issue title. The correct link for justifying closing is this one, which IS in the source tree:
> 
> https://github.com/audacity/audacity/blob/master/INSTALL

Fair enough. I had assumed that (upper case) INSTALL in the root directory was so obvious that he couldn't  mean that. Thanks for the correction.

**Comment by BenBE** [comment]
Somehow missed that when looking through things …
